File: app/src/main/java/ru/serjik/utils/HexUtils.java
package ru.serjik.utils;

public class HexUtils {
    public static final float SQRT3_OVER_2 = (float) (Math.sqrt(3.0d) / 2.0d);
    public static final byte[] DIRECTION_Q = {1, 1, 0, -1, -1, 0};
    public static final byte[] DIRECTION_R = {-1, 0, 1, 1, 0, -1};
    public static final float[] DIRECTION_ANGLES = {120.0f, 180.0f, 240.0f, 300.0f, 0.0f, 60.0f};

    public static float hexY(int r) {
        return r * 0.75f;
    }

    public static float hexX(int q, int r) {
        return (q * SQRT3_OVER_2) + (r * SQRT3_OVER_2 * 0.5f);
    }
}

File: app/src/main/java/ru/serjik/utils/HexUtils.kt
package ru.serjik.utils

import kotlin.math.sqrt

object HexUtils {
    val SQRT3_OVER_2: Float = (sqrt(3.0) / 2.0).toFloat()

    val DIRECTION_Q: ByteArray = byteArrayOf(1, 1, 0, -1, -1, 0)

    val DIRECTION_R: ByteArray = byteArrayOf(-1, 0, 1, 1, 0, -1)

    val DIRECTION_ANGLES: FloatArray = floatArrayOf(120.0f, 180.0f, 240.0f, 300.0f, 0.0f, 60.0f)

    fun hexY(r: Int): Float = r * 0.75f

    fun hexX(q: Int, r: Int): Float = (q * SQRT3_OVER_2) + (r * SQRT3_OVER_2 * 0.5f)
}
